pkg/autoscaler/autoscaler/util.go

```diff
@@ -0,0 +1,133 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package autoscaler
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/pingcap/tidb-operator/pkg/apis/pingcap/v1alpha1"
+	"github.com/pingcap/tidb-operator/pkg/label"
+	operatorUtils "github.com/pingcap/tidb-operator/pkg/util"
+	appsv1 "k8s.io/api/apps/v1"
+	autoscalingv2beta2 "k8s.io/api/autoscaling/v2beta2"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/utils/pointer"
+)
+
+// checkStsAutoScalingPrerequisites would check the sts status to ensure wouldn't happen during
+// upgrading, scaling
+func checkStsAutoScalingPrerequisites(set *appsv1.StatefulSet) bool {
+	if operatorUtils.IsStatefulSetUpgrading(set) {
+		return false
+	}
+	if operatorUtils.IsStatefulSetScaling(set) {
```

Comment:
This can be avoided by scalingxxInterval?

Reply:
It can't be. We wouldn't auto-scaling during manually scaling.